lib/problems.js

```diff
@@ -1,9 +1,30 @@
+const Stack = require('./stack')
 /*
-Time Complexity: ?
-Space Complexity: ?
+Time Complexity: O(n^2) - the stack.pop() method invoked has O(n) time complexity and it is used inside a for loop.
+Space Complexity: O(n) - we keep pushing values to the stack for a string of length n.
 */
 const balanced = (str) => {
```

Comment:
👍 I'm pretty sure the `pop` method has O(1) time complexity leading your solution to O(n) time complexity overall.